bot/cogs/fun.py

```diff
@@ -193,24 +206,87 @@ def __init__(self, bot: Bot):
         Returns a random quotation from the #quotes channel.
         A user can be specified to return a random quotation from that user.
         """
+        conn = await asyncpg.connect()
         quote_channel = self.bot.get_channel(QUOTES_CHANNEL_ID)
-        quotes = self.bot.quotes
 
         if member is None:
-            message_id = choice(quotes[choice(list(quotes.keys()))])
+            message_id = await conn.fetchval(
+                "select quote_id from quotes order by random() limit 1;"
+            )  # fetchval returns first value by default
         else:
-            user_quotes = quotes[f'{member.name}#{member.discriminator}']
-            if not user_quotes:
-                await ctx.send("No quotes from that user.")
+            message_id = await conn.fetchval(
+                "select quote_id from quotes where author_id=$1 order by random() limit 1;",
+                member.id
+            )
+            if message_id is None:
+                await ctx.send("No quotes for that user.")
                 return
-            message_id = choice(user_quotes)
 
+        await conn.close()
         message = await quote_channel.get_message(message_id)
-        await ctx.send(embed=message.embeds[0])
+        if message.author.id == QUOTES_BOT_ID:
```

Comment:
Line 227 to 235 don't account for all possible types of quote, here's a suggestion but I'm not a perfect programmer so feel free to chip in (This should account for embeds and multiple attachments and embedding attachments where possible)
```suggestion
        embed = None
        content = message.content
        attachment_urls = [attachment.url for attachment in message.attachments]

        if message.embeds: 
            embed = message.embeds[0]
        elif len(attachments) == 1:
            image_url = attachments.pop(0)
            embed = Embed()
            embed.set_image(url=image_url)

        for url in attachments_urls:
            content += "\n" + url

        await ctx.send(content, embed=embed)
```